cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -189,126 +181,141 @@ public CruiseControlApiImpl(Vertx vertx, int idleTimeout) {
+                    response.bodyHandler(buffer -> {
+                        JsonObject json = buffer.toJsonObject();
+                        if (json.containsKey(CC_REST_API_ERROR_KEY)) {
+                            result.fail(json.getString(CC_REST_API_ERROR_KEY));
+                        } else {
+                            CruiseControlResponse ccResponse = new CruiseControlResponse(userTaskID, json);
+                            result.complete(ccResponse);
+                        }
+                    });
+
+                } else {
+                    result.fail(new CruiseControlRestException(
+                            "Unexpected status code " + response.statusCode() + " for GET request to " +
+                                    host + ":" + port + path));
+                }
+            }).exceptionHandler(t -> httpExceptionHandler(result, t));
+
+            if (idleTimeout != HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS) {
+                request.setTimeout(idleTimeout * 1000);
+            }
+
+            request.end();
+        });
     }
 
     private void httpExceptionHandler(Promise<? extends CruiseControlResponse> result, Throwable t) {
         if (t instanceof TimeoutException) {
             // Vert.x throws a NoStackTraceTimeoutException (inherits from TimeoutException) when the request times out
-            // goint to catch and raise a TimeoutException instead
+            // so we catch and raise a TimeoutException instead
             result.fail(new TimeoutException(t.getMessage()));
         } else if (t instanceof ConnectException) {
             // Vert.x throws a AnnotatedConnectException (inherits from ConnectException) when the request times out
-            // goint to catch and raise a ConnectException instead
+            // so we catch and raise a ConnectException instead
             result.fail(new ConnectException(t.getMessage()));
         } else {
             result.fail(t);
         }
     }
+
+    /**
+     * Perform the given operation, which completes the promise, using an HTTP client instance,
+     * after which the client is closed and the future for the promise returned.
+     * @param operation The operation to perform.
+     * @param <T> The type of the result
+     * @return A future which is completed with the result performed by the operation
+     */
+    private <T> Future<T> withHttpClient(BiConsumer<HttpClient, Promise<T>> operation) {
+        HttpClient httpClient = vertx.createHttpClient(new HttpClientOptions().setLogActivity(HTTP_CLIENT_ACTIVITY_LOGGING));
+        Promise<T> promise = Promise.promise();
+        operation.accept(httpClient, promise);
+        return promise.future().compose(
+            result -> {
+                httpClient.close();
+                return Future.succeededFuture(result);
+            },
+            error -> {
+                httpClient.close();
+                return Future.failedFuture(error);
+            });
+    }
```

Comment:
This looks identical to the one Tom has in `KafkaConnectApiImpl`. Can we move it to some `Util` or `HttpUtil` class and share it? Or does it actually use something from the class and cannot be static?

Reply:
That is a good idea. I have moved the method to an external util class. Should I edit the Connect client to use it in this same PR or open another after this is merged?

Reply:
I think you can do it in this PR.

Reply:
Done